<?xml version="1.0"?>
<launch>
  <!-- Action namespace -->
  <arg name="action_name" default="generate_grasps"/>
  <arg name="frame_id" value="panda_link0" doc="frame of the grasps sent to MTC"/>
  <arg name="point_cloud_topic" value="/camera/depth/color/points" doc="point cloud topic name"/>

  <!-- file paths -->
  <arg name="path_to_pcd_file" value="$(find deep_grasp_demo)/config/cylinder.pcd" doc="path to point cloud file"/>
  <arg name="path_to_gpd_config" value="$(find deep_grasp_demo)/config/gpd_config.yaml" doc="path to GPD config and gripper params file"/>

  <!-- Run MTC pick and place -->
  <node name="deep_grasp_demo" pkg="deep_grasp_task" type="deep_grasp_demo" output="screen">
      <param name="execute" value="true" />
      <param name="action_name" value="$(arg action_name)"/>
      <rosparam command="load" file="$(find gqcnn_demo)/config/panda_object.yaml" />
  </node>

  <!-- Handles the GPD lib -->
  <node name="gpd_action_server" pkg="deep_grasp_demo" type="gpd_action_server" output="screen">
    <rosparam param="view_point">[0.0, 0.0, 0.0]</rosparam>
    <param name="action_name" value="$(arg action_name)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="point_cloud_topic" value="$(arg point_cloud_topic)"/>
    <!-- <param name="path_to_pcd_file" value="$(arg path_to_pcd_file)"/> -->
    <param name="path_to_gpd_config" value="$(arg path_to_gpd_config)"/>
    <rosparam command="load" file="$(find deep_grasp_demo)/config/calib/camera.yaml" />
  </node>
</launch>
